AnkiDroid/src/main/java/com/ichi2/anki/AddonsBrowser.java

```diff
@@ -0,0 +1,425 @@
+
+        // Add a home button to the actionbar
+        getSupportActionBar().setHomeButtonEnabled(true);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setTitle(getString(R.string.reviewer_addons));
+        showBackIcon();
+
+        npmUtility = new AddonsNpmUtility(AddonsBrowser.this, this);
+
+        addonsDownloadButton = findViewById(R.id.addons_download_message_ll);
+        addonsDownloadButton.setOnClickListener(v -> {
+            openUrl(Uri.parse(getString(R.string.ankidroid_js_addon_npm_search_url)));
+        });
+
+        addonsListRecyclerView = findViewById(R.id.addons);
+        addonsListRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(this, DividerItemDecoration.VERTICAL);
+        addonsListRecyclerView.addItemDecoration(dividerItemDecoration);
+
+        String currentAnkiDroidDirectory = CollectionHelper.getCurrentAnkiDroidDirectory(this);
+        addonsHomeDir = new File(currentAnkiDroidDirectory, "addons");
+
+        preferences = AnkiDroidApp.getSharedPrefs(this);
+
+        listAddonsFromDir(addonType);
+    }
+
+
+    @Override
+    public String getSubtitleText() {
+        return getResources().getString(R.string.addons);
+    }
+
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        finishWithAnimation(RIGHT);
+
+        if (downloadDialog.isShowing()) {
+            downloadDialog.dismiss();
+        }
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuItem installAddon, getMoreAddons, reviewerAddons;
+
+        getMenuInflater().inflate(R.menu.addon_browser, menu);
+        installAddon = menu.findItem(R.id.action_install_addon);
+        getMoreAddons = menu.findItem(R.id.action_get_more_addons);
+        reviewerAddons = menu.findItem(R.id.action_addon_reviewer);
+
+        reviewerAddons.setOnMenuItemClickListener(item -> {
+            getSupportActionBar().setTitle(getString(R.string.reviewer_addons));
+            listAddonsFromDir(addonType);
+            return true;
+        });
+
+        downloadDialog = new Dialog(this);
+
+        installAddon.setOnMenuItemClickListener(item -> {
+
+            downloadDialog.setCanceledOnTouchOutside(true);
+            downloadDialog.setContentView(R.layout.addon_install_from_npm);
+
+            EditText downloadEditText = downloadDialog.findViewById(R.id.addon_download_edit_text);
+            Button downloadButton = downloadDialog.findViewById(R.id.addon_download_button);
+
+            downloadButton.setOnClickListener(v -> {
+                String npmAddonName = downloadEditText.getText().toString();
+
+                // if string is:  npm i ankidroid-js-addon-progress-bar
+                if (npmAddonName.startsWith("npm i")) {
+                    npmAddonName = npmAddonName.replaceAll("npm i", "");
```

Comment:
We only want a substring, not replacing all instances

```suggestion
                    npmAddonName = npmAddonName.substring("npm i".length());
```